lib/sqlalchemy/orm/strategy_options.py

```diff
@@ -232,7 +232,16 @@ def _generate_path(
                 else:
                     return None
             else:
-                attr = found_property = attr.property
```

Comment:
OK, how about, "if not isinstance(attr, MapperProperty):  raise ...("Expected %s to be a mapped column; instaed got %r object")

that way if they have any other kind of attribute or descriptor there we still raise something meaningful

Reply:
that's a much better idea.